remote_attestation/java/src/com/google/oak/remote_attestation/ClientHandshaker.java

```diff
@@ -91,88 +75,104 @@ public ClientHandshaker(byte[] expectedTeeMeasurement) {
+
+      state = State.ExpectingServerIdentity;
+      return serializedClientHello;
+    } catch (Exception e) {
+      state = state.Aborted;
+      throw e;
     }
-
-    // Random vector sent in messages for preventing replay attacks.
-    byte[] random = new byte[REPLAY_PROTECTION_ARRAY_LENGTH];
-    new Random().nextBytes(random);
-
-    // Create client hello message.
-    Message.ClientHello clientHello = new Message.ClientHello(random);
-
-    // Update current transcript.
-    byte[] serializedClientHello = clientHello.serialize();
-    appendTranscript(serializedClientHello);
-
-    state = state.transition();
-    return serializedClientHello;
   }
 
   /**
    * Responds to {@code ServerIdentity} message by creating a {@code ClientIdentity} message and
    * derives session keys for encrypting/decrypting messages from the server.
    * {@code ClientIdentity} message contains an ephemeral public key for negotiating session keys.
    *
-   * Transitions {@code ClientHandshaker} state from {@code Attesting} to {@code Attested} state.
+   * Transitions {@code ClientHandshaker} state from {@code State.ExpectingServerIdentity} to
+   * {@code State.Attested} state.
    */
   public byte[] processServerIdentity(byte[] serializedServerIdentity)
       throws IllegalStateException, IOException, GeneralSecurityException {
-    if (state != HandshakerState.ExpectingServerIdentity) {
-      throw new IllegalStateException(
-          "ClientHandshaker is not in the ExpectingServerIdentity state");
-    }
-
-    Message.ServerIdentity serverIdentity =
-        Message.ServerIdentity.deserialize(serializedServerIdentity);
-
-    // Update current transcript.
-    // Transcript doesn't include transcript signature from the server identity message.
-    ServerIdentity serverIdentityNoSignature = new Message.ServerIdentity(
-        serverIdentity.getEphemeralPublicKey(), serverIdentity.getRandom(),
-        serverIdentity.getSigningPublicKey(), serverIdentity.getAttestationInfo());
-    byte[] serializedServerIdentityNoSignature = serverIdentityNoSignature.serialize();
-    appendTranscript(serializedServerIdentityNoSignature);
-
-    // Verify server transcript signature.
-    byte[] transcriptHash = Hashing.sha256().hashBytes(transcript).asBytes();
-    SignatureVerifier verifier = new SignatureVerifier(serverIdentity.getSigningPublicKey());
-    if (!verifier.verify(transcriptHash, serverIdentity.getTranscriptSignature())) {
-      throw new VerifyException("Couldn't verify server transcript signature");
-    }
-
-    // Verify server attestation info.
-    if (!verifyAttestationInfo(serverIdentity.getAttestationInfo())) {
-      throw new VerifyException("Couldn't verify attestation info");
+    try {
+      if (state != State.ExpectingServerIdentity) {
+        throw new IllegalStateException(
+            "ClientHandshaker is not in the ExpectingServerIdentity state");
+      }
+
+      Message.ServerIdentity serverIdentity =
```

Comment:
Could Message be dropped? 
```suggestion
     ServerIdentity serverIdentity =
```

Reply:
It's a `static` class member, it can't be dropped

Reply:
You have this import line: `import com.google.oak.remote_attestation.Message.ServerIdentity;`, so I believe `ServerIdentity` works too. But it is better to be consistent. If it is now `Message.ServerIdentity` everywhere, remove the import.

Reply:
Done

Reply:
I think this import was added by vscode automatically